app/controllers/fhir/r4/api_controller.rb

```diff
@@ -29,6 +29,7 @@ class Fhir::R4::ApiController < ApplicationApiController
   end
```

Comment:
(arbitrarily commenting on this line so we can have a thread... thanks github)

Just wanted to get your current thinking regarding the possibility of separating out controllers for each resource type. Not trying to unnecessarily create more work for you, especially because I think this works pretty well as is. I just think it would be more railsy and could make the API code more maintainable and/or approachable, as opposed to the long (though now shorter) controller methods with big switch statements.

I'm imagining the shared code/helper methods defined here going into the `ApplicationApiController` super class (and/or helper class(es)). So pretty much all of the changes here would still be relevant and necessary for the approach I'm describing. So I'm approving this as is, but still want to pose the question as a potential future task.

Reply:
Yeah I wanted to investigate that as a part of this task, but in the end other things of a bit higher priority ended up popping up. I've made a separate task to properly try that out here: https://tracker.codev.mitre.org/browse/SARAALERT-1526.